storage/CompoundAction/Formatter/Md.cc

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (c) 2017 SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+
+#include <boost/algorithm/string.hpp>
+
+#include "storage/CompoundAction/Formatter/Md.h"
+#include "storage/Filesystems/Swap.h"
+
+
+
+namespace storage
+{
+
+    CompoundAction::Formatter::Md::Md( const CompoundAction::Impl* compound_action ):
+	CompoundAction::Formatter( compound_action, "Md" ),
+	md( to_md( compound_action->get_target_device() ) )
+    {
+    }
+
+
+    Text
+    CompoundAction::Formatter::Md::text() const
+    {
+	if ( deleting() )
+	    return delete_text();
+
+	else if ( formatting() && is_swap( get_created_filesystem() ) )
+	{
+	    if ( encrypting() )
+		return create_encrypted_with_swap_text();
+
+	    else
+		return create_with_swap_text();
+	}
+
+	else
+	{
+	    if ( creating() && encrypting() && formatting() && mounting() )
+		return create_encrypted_with_fs_and_mount_point_text();
+
+	    else if ( creating() && encrypting() && formatting() )
+		return create_encrypted_with_fs_text();
+
+	    else if ( creating() && encrypting() )
+		return create_encrypted_text();
+
+	    else if ( creating() && formatting() && mounting() )
+		return create_with_fs_and_mount_point_text();
+
+	    else if ( creating() && formatting() )
+		return create_with_fs_text();
+
+	    else if ( creating() )
+		return create_text();
+
+	    else if ( encrypting() && formatting() && mounting() )
+		return encrypted_with_fs_and_mount_point_text();
+
+	    else if ( encrypting() && formatting() )
+		return encrypted_with_fs_text();
+
+	    else if ( encrypting() )
+		return encrypted_text();
+
+	    else if ( formatting() && mounting() )
+		return fs_and_mount_point_text();
+
+	    else if ( formatting() )
+		return fs_text();
+
+	    else if ( mounting() )
+		return mount_point_text();
+
+	    else
+		return default_text();
+	}
+    }
+
+
+    Text
+    CompoundAction::Formatter::Md::devices_text() const
+    {
+	std::vector<const BlkDevice *> devices = md->get_devices();
+	std::sort( devices.begin(), devices.end(),
+		   [](const BlkDevice * a, const BlkDevice * b) -> bool
+		       {
+			   return a->get_name() < b->get_name();
```

Comment:
This should use the special compare function for BlkDevice, BlkDevice::compare_by_name(). See testsuite/sorting/blk-device1.cc. Sorry, you do not even need a lambda then.